src/visualizers/VisualizerHistogram.ts

```diff
@@ -0,0 +1,211 @@
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding index.
+    orderedFactorArray(): number[] {
+        const orderedFactorArray = []
+        const factorArray = this.factorArray()
+        for (
+            let i = 0;
+            i < natlog(this.largestValue()) / natlog(2) + 1;
+            i++
+        ) {
+            orderedFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            orderedFactorArray[factorArray[i]]++
+        }
+
+        return orderedFactorArray
+    }
+
+    // Create an array to change the bin size of the
+    // histogram to what the user asked for.
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const orderedFactorArray = this.orderedFactorArray()
+        for (
+            let i = 0;
+            i < Math.ceil(orderedFactorArray.length / this.binSize);
+            i++
+        ) {
+            binFactorArray.push(0)
+        }
+
+        let index = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            for (let j = 0; j < this.binSize; j++) {
+                if (index + j < orderedFactorArray.length) {
+                    binFactorArray[i] += orderedFactorArray[index + j]
+                }
+            }
+            index += this.binSize
+        }
+
+        return binFactorArray
+    }
+
+    // Create a number that represents how
+    // many pixels wide each bin should be.
+    binWidth(): number {
+        const binFactorArray = this.binFactorArray()
+        let binWidth = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] != 0) {
+                binWidth = i
+            }
+        }
+        return 750 / (binWidth + 1)
+    }
+
+    // Create a number that represents how many pixels high
+    // each increase of one in the bin array should be.
+    height(): number {
+        const binFactorArray = this.binFactorArray()
+        let height = binFactorArray[0]
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] > height) {
+                height = binFactorArray[i]
+            }
+        }
+        return 750 / height
+    }
+
+    draw() {
+        this.sketch.background(176, 227, 255)
+        const height = this.height()
+        const binWidth = this.binWidth()
+        const binFactorArray = this.binFactorArray()
+        this.sketch.line(40, 10, 40, 800)
+        this.sketch.line(0, 760, 790, 760)
+
+        for (let i = 0; i < binFactorArray.length; i++) {
+            this.sketch.rect(
+                40 + binWidth * i,
+                Number(760 - height * binFactorArray[i]),
+                binWidth,
+                height * binFactorArray[i]
+            )
+            if (this.binSize != 1) {
+                this.sketch.text(
+                    i
+                        + this.binSize * i
+                        - i
+                        + ' - '
```

Comment:
This expression appears to have an 'i' and a '-i' in it, which seem like they should cancel...

Reply:
Substituted a more specific comment for this revision, so resolving this one.